### PR TITLE
Create canadian-journal-of-soil-sciences.csl

### PR DESCRIPTION
**Pull request by libora**
Style for Canadian Journal of Soil Science based on style for Canadian Journal of Fisheries and Aquatic Sciences but there is a lot of changes due instruction for authors (Operations Manual) of CJSS

### CODE REVIEW

File: canadian-journal-of-soil-sciences.csl [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+  </macro>
+  <macro name="locators">
+    <choose>
+      <if type="article-journal article-magazine article-newspaper" match="any">
+        <text variable="container-title" form="short"/>
+        <text variable="volume" font-weight="bold" prefix=" "/>
+        <text variable="page" prefix=": "/>
+      </if>
+      <else-if type="chapter paper-conference" match="any">
+        <group delimiter=". ">
```
Comment (adam3smith) [inline canadian-journal-of-soil-sciences.csl]: This group isn't doing anything. Remove

File: canadian-journal-of-soil-sciences.csl [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+            <text macro="secondary-contributors"/>
+            <group delimiter=", ">
+              <text variable="container-title"/>
+              <text macro="edition"/>
+            </group>
+          </group>         
+        </group>
+      </else-if>
+      <else-if type="bill book graphic legal_case legislation motion_picture report song" match="any">
+        <group delimiter=". ">
```
Comment (adam3smith) [inline canadian-journal-of-soil-sciences.csl]: Same here

File: canadian-journal-of-soil-sciences.csl [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+      <text variable="locator" prefix=" "/>
+    </group>
+  </macro>
+  <citation et-al-min="3" et-al-use-first="1" disambiguate-add-year-suffix="true" collapse="year">
+    <sort>
+      <key variable="issued"/>
+      <key macro="author"/>
+    </sort>
+    <layout prefix="(" suffix=")" delimiter=", ">
+      <group delimiter=" ">
```
Comment (adam3smith) [inline canadian-journal-of-soil-sciences.csl]: Just making sure this is on purpose. Only spare as delimiter seems odd

**Comment by adam3smith** [comment]
A couple of miss inline. Looks great otherwise

**Comment by adam3smith** [comment]
A couple of nits inline. Looks great otherwise

File: canadian-journal-of-soil-sciences.csl (hunk 3 above)
Comment (libora) [inline canadian-journal-of-soil-sciences.csl]: @adam3smith Sorry I do not understand what you exactly mean. Tag citation I am not changed and it is the same as in your template style.

Comment (adam3smith) [inline canadian-journal-of-soil-sciences.csl]: As long as (Smith 1776 p. 5) is right, this is fine. Just making sure that's the case

Comment (libora) [inline canadian-journal-of-soil-sciences.csl]: There is nothing about locators in text citation in guidance for authors. The example in guidance for authors is "Several authors (Thomas 1982; Alex et al. 1983; Smith 1990) have shown..."

File: canadian-journal-of-soil-sciences.csl [hunk 4]
```diff
@@ -0,0 +1,222 @@
+<?xml version="1.0" encoding="utf-8"?>
+<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" default-locale="en-US" version="1.0" demote-non-dropping-particle="sort-only">
+  <info>
+    <title>Canadian Journal of Soil Science</title>
+    <id>http://www.zotero.org/styles/canadian-journal-of-soil-sciences</id>
```
Comment (rmzelle) [inline canadian-journal-of-soil-sciences.csl]: This should be `<id>http://www.zotero.org/styles/canadian-journal-of-soil-sciences</id>` ("science", not "sciences"). Idem for file name and "self" link.

http://www.nrcresearchpress.com/journal/cjss

Comment (libora) [inline canadian-journal-of-soil-sciences.csl]: Sorry, my mistake.

**Comment by libora** [comment]
@rmzelle @adam3smith Hallo guys, do you have any other comments?

**Comment by rmzelle** [comment]
Looks good to me. Will this one also have dependents?

And in general, it's always good to check whether Canadian journals use British English ("en-GB") or US English ("en-US") spelling.

**Comment by libora** [comment]
Yes this style has two dependents. I do not change spelling, in this stale is the same spelling as in template.

**Comment by rmzelle** [comment]
> I do not change spelling, in this stale is the same spelling as in template.

I meant, if you see any indications that the style uses British spelling, like placing punctuation outside of quotes (e.g. `"Title", edited by A` (British) vs. `"Title," edited by A` (US)), you can set the `default-locale` to "en-GB". The date format is also a giveaway (dd-mm-yyyy for British, mm-dd-yyyy for US).

**Comment by libora** [comment]
No, I did not control it, this is a little higher level than my knowledge of English :(

**Comment by rmzelle** [comment]
Okay, that's fine :).

I just checked the style guide, and the only full dates I see have the format "[1998 Oct. 01]", so that's custom anyways. The guide also says:

> Use American (refer to Webster's New International
Dictionary) or British (refer to the Oxford English Dictionary)
spelling consistently throughout the manuscript.

So "en-US" should be good.

**Comment by libora** [comment]
Thanks. I am in bus now and have no chance to check style guidance yet.

**Comment by rmzelle** [comment]
Thanks!

**Pull request merged**